readthedocs/embed/v3/views.py

```diff
@@ -0,0 +1,318 @@
+        external_domain = parsed_url.netloc
+        if external and external_domain:
+            allowed_domain = False
+            for domain in settings.RTD_EMBED_API_EXTERNAL_DOMAINS:
+                if re.match(domain, external_domain):
+                    allowed_domain = True
+                    break
+
+            if not allowed_domain:
+                log.info('Domain not allowed. domain=%s url=%s', external_domain, url)
+                return Response(
+                    {
+                        'error': (
+                            'External domain not allowed. '
+                            f'domain={external_domain}'
+                        )
+                    },
+                    status=status.HTTP_400_BAD_REQUEST,
+                )
+
+            # Check rate-limit for this particular domain
+            cache_key = f'embed-api-{external_domain}'
+            cache.get_or_set(cache_key, 0, timeout=settings.RTD_EMBED_API_DOMAIN_RATE_LIMIT_TIMEOUT)
+            cache.incr(cache_key)
+            if cache.get(cache_key) > settings.RTD_EMBED_API_DOMAIN_RATE_LIMIT:
+                log.warning('Too many requests for this domain. domain=%s', external_domain)
+                return Response(
+                    {
+                        'error': (
+                            'Too many requests for this domain. '
+                            f'domain={external_domain}'
+                        )
+                    },
+                    status=status.HTTP_429_TOO_MANY_REQUESTS,
+                )
+
+        # NOTE: we could validate the fragment if we want. It must contain at
+        # least one character, cannot start with a number, and must not contain
+        # whitespaces (spaces, tabs, etc.).
+        fragment = parsed_url.fragment
+
+        content_requested = self._get_content_by_fragment(
+            url,
+            fragment,
+            external,
+            doctool,
+            doctoolversion,
+        )
+        if not content_requested:
+            log.warning('Identifier not found. url=%s', url)
+            return Response(
+                {
+                    'error': (
+                        "Can't find content for section: "
+                        f"url={url} fragment={fragment}"
+                    )
+                },
+                status=status.HTTP_404_NOT_FOUND
+            )
+
+        response = {
+            'url': url,
+            'fragment': fragment,
+            'content': clean_links(
+                content_requested,
+                url,
+                html_raw_response=True,
+            ),
+            'external': external,
+        }
+
+        if not external:
+            response.update({
+                'project': self.unresolved_url.project.slug,
+                'version': self.unresolved_url.version_slug,
+                'language': self.unresolved_url.language_slug,
+                'path': self.unresolved_url.filename,
```

Comment:
Honestly, I haven't used this data, but I got used to having it there from the APIv2 and it gave me a good understanding of "what the endpoint understood from the URL that I passed".

I'm fine removing it for now because I don't like having different responses depending on external/internal neither. We can come back here and add them if we found them useful for users.